sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/oidc/OidcClient.java

```diff
@@ -0,0 +1,159 @@
+package org.sdase.commons.client.jersey.oidc;
+
+import static org.sdase.commons.client.jersey.proxy.ApiClientInvocationHandler.createProxy;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.Form;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.glassfish.jersey.client.proxy.WebResourceFactory;
+import org.sdase.commons.client.jersey.ClientFactory;
+import org.sdase.commons.client.jersey.oidc.model.OidcResult;
+import org.sdase.commons.client.jersey.oidc.model.OidcState;
+import org.sdase.commons.client.jersey.oidc.rest.OpenIdDiscoveryApi;
+import org.sdase.commons.client.jersey.oidc.rest.model.OpenIdDiscoveryResource;
+import org.sdase.commons.client.jersey.oidc.rest.model.TokenResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Service that can retrieve a new access token. */
+public class OidcClient {
+
+  private static final String CACHE_KEY = "oidcResultKey";
+  private static final long NANOSECOND_CONVERSION_FACTOR = 1000000000;
```

Comment:
NIT: A little bit more readability:
```suggestion
  private static final long NANOSECOND_CONVERSION_FACTOR = 1_000_000_000;
```